File: guides/micronaut-mongodb-synchronous/java/src/test/java/example/micronaut/FruitClient.java
package example.micronaut;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.http.HttpStatus;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Post;
import io.micronaut.http.client.annotation.Client;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import java.util.List;

@Client("/fruits")
public interface FruitClient {

    @Post
    @NonNull
    HttpStatus save(@NonNull @NotNull @Valid Fruit fruit);

    @NonNull
    @Get
    List<Fruit> findAll();
}

File: guides/micronaut-mongodb-asynchronous/kotlin/src/test/kotlin/example/micronaut/FruitClient.kt
package example.micronaut

import io.micronaut.http.HttpStatus
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Post
import io.micronaut.http.client.annotation.Client
import jakarta.validation.Valid

@Client("/fruits")
interface FruitClient {

    @Post
    fun save(@Valid fruit: Fruit): HttpStatus

    @Get
    fun findAll(): List<Fruit>
}
